### PR TITLE
Add functional task to install Pangeo helm chart and config templates

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  command: helm repo update
+
+- name: Copy Helm jupyter config to host
+  template:
+    src: jupyter_config.yaml.j2
+    dest: /root/helm/jupyter_config.yaml
+
+- name: Copy Helm jupyter config to host
+  template:
+    src: secret_config.yaml.j2
```
Comment (brtkwr) [inline tasks/main.yml]: chmod 600

File: templates/jupyter_config.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+jupyterhub:
+  singleuser:
+    image:
+      name: pangeo/pangeo-notebook
+      tag: 2019.01.31
+    startTimeout: 1000
+    extraEnv:
+      EXTRA_PIP_PACKAGES: >-
+      GCSFUSE_BUCKET: pangeo-data
```
Comment (brtkwr) [inline templates/jupyter_config.yaml.j2]: remove

File: templates/jupyter_config.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# file: jupyter_config.yaml
+
+jupyterhub:
+  singleuser:
+    image:
+      name: pangeo/pangeo-notebook
+      tag: 2019.01.31
+    startTimeout: 1000
+    extraEnv:
+      EXTRA_PIP_PACKAGES: >-
```
Comment (brtkwr) [inline templates/jupyter_config.yaml.j2]: remove

File: templates/secret_config.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# file: secret_config.yaml
+
+jupyterhub:
+  proxy:
+    secretToken: 8ab358498cadf3e05db75341c3ba9b41d65e565433a6ec17020afe31f2dd7806
```
Comment (brtkwr) [inline templates/secret_config.yaml.j2]: use ansible_facts, look at awx

Comment (brtkwr) [reply templates/secret_config.yaml.j2]: rabbitmq

File: tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
+---
+
+- name: "Copy Helm {{ item }} to host"
+  template:
```
Comment (oneswig) [inline tasks/main.yml]: Set a mode, eg 0600, in particular secret_config looks a bit sensitive?

**Review by oneswig — COMMENTED** [review]
What's with the secret token committed?

**Review by mkjpryor-stfc — CHANGES_REQUESTED** [review]
Looks pretty good. I just have one comment really.

The `pangeo_upgrade` flag is not required - according to the playbook compatibility document, Pangeo should be upgraded if a different `pangeo_version` is specified to the installed version. I think this would also allow you to get away with a single `helm upgrade --install` task handling initial installation and upgrading.

(If you look at this PR - https://github.com/cedadev/k8s-ansible/pull/7 - it actually removes the `upgrade_kubernetes` flag from `k8s-ansible` and replaces it with a check whether the requested version matches the currently installed one.

**Comment by brtkwr** [comment]
Thanks for the feedback @mkjpryor-stfc , I have addressed your comment :)

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  when: pangeo_state == "present"
+
+- name: Check if the chart exists
+  command: "helm ls {{ pangeo_namespace }} --output json"
+  register: output
+  changed_when: false
+
+- name: Set fact about the output
+  set_fact:
+    current_release: "{{ ({} if output.stdout == '' else (output.stdout | from_json)).get('Releases', [{}]) | first }}"
```
Comment (mkjpryor-stfc) [inline tasks/main.yml]: Could this be made cleaner with the `json_query` filter?

https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#json-query-filter

**Review by mkjpryor-stfc — CHANGES_REQUESTED** [review]
Other than that, looks good.

**Pull request merged**